src/Compilers/CSharp/Portable/Symbols/DiscardedSymbol.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal class DiscardedSymbol : Symbol, IDiscardedSymbol
+    {
+        private readonly TypeSymbol _type;
+
+        public DiscardedSymbol(TypeSymbol type)
+        {
+            Debug.Assert((object)type != null);
+            _type = type;
+        }
+
+        ITypeSymbol IDiscardedSymbol.Type => _type;
+        public TypeSymbol Type => _type;
+
+        public override Symbol ContainingSymbol => null;
+        public override Accessibility DeclaredAccessibility => Accessibility.NotApplicable;
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray<SyntaxReference>.Empty;
+        public override bool IsAbstract => false;
+        public override bool IsExtern => false;
+        public override bool IsImplicitlyDeclared => true;
+        public override bool IsOverride => false;
+        public override bool IsSealed => false;
+        public override bool IsStatic => true;
+        public override bool IsVirtual => false;
+        public override SymbolKind Kind => SymbolKind.Discarded;
+        public override ImmutableArray<Location> Locations => ImmutableArray<Location>.Empty;
```

Comment:
We have not been able to imagine any use cases for these symbols providing a location. This version of the API was reviewed by about a half-dozen people, so if we make a change here it should be reviewed again.